core/src/main/java/org/fao/geonet/util/XslUtil.java

```diff
@@ -735,6 +744,46 @@ public static String encodeForURL(String str) {
         }
     }
 
+    /**
+     *  To get the xml content of an url
+     *  It supports the usage of a proxy
+        * @param surl
+        * @return
+     */
+    public static Node getUrlContent(String surl) {
+
+        Node res = null;
+        InputStream is = null;
+
+        ServiceContext context = ServiceContext.get();
+
+        try {
+            URL url = new URL(surl);
+            URLConnection conn = Lib.net.setupProxy(context, url);
+
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            DocumentBuilder db = dbf.newDocumentBuilder();
+
+            is = conn.getInputStream();
+            res = db.parse(is);
+
+        } catch (Throwable e) {
+            e.printStackTrace();
```

Comment:
Would be better to remove the `printStackTrace` and add the exception as parameter in `Log.error`